### ISSUE TITLE
RLS: geopandas 0.3 release

### ISSUE THREAD

**Issue by jdmcbr** [geopandas/geopandas#470]
@jorisvandenbossche @kjordahl I think it would be good to target a release soon. Though perhaps it should wait for #467, given the rapid progress being made there. Should we wait on that? Or on any other pending PRs?

Disregarding any pending pull requests that might be included, a rough draft changelog since ``0.2.1``:

Improvements:
* improved plotting performance using ``matplotlib.collections`` (#267)
* return empty data frame rather than raising an error when performing a spatial join with non overlapping geodataframes (#335) [note, this fix for #251 seems like it predates the 0.2.1 release, but still see the same error on a few systems when installed from 0.2.1, but not when installed from master]
* enable ``sjoin`` on non-integer-index GeoDataFrames (#422)
* provide access to x/y coordinates as attributes for Point GeoSeries (#383)

Bug fixes:
* fixed ``fiona.filter`` results when ``bbox`` is not None (#372)
* fixed ``dissolve`` to retain CRS (#389)

**Comment by mrocklin** [comment]
> Though perhaps it should wait for #467, given the rapid progress being made there. Should we wait on that? 

I actually recommend releasing before any of the Cython stuff lands.  I think it's good to release before a period of heavy churn so that existing improvements in master aren't necessarily bundled with possibly breaking changes.

**Comment by jorisvandenbossche** [comment]
> I actually recommend releasing before any of the Cython stuff lands

Yes, that is also what I had in mind. And we can certainly use a release (thanks for @jdmcbr for pinging for that!). But let's call it 0.3, as we have some nice improvements.

**Comment by jorisvandenbossche** [comment]
@jdmcbr The above changelog you list looks already quite accurate. Would you like to make a PR to add that to the `CHANGELOG` file?

**Comment by jorisvandenbossche** [comment]
Some things I *would* like to include before releasing (but not all are release critical), are the following:

- fix the tests for matplotlib 2.0 (https://github.com/geopandas/geopandas/issues/278)
- A decision / implementation on better plotting defaults (https://github.com/geopandas/geopandas/issues/318)
- Fix some plots in the docs (https://github.com/geopandas/geopandas/issues/466)
- the sjoin failures and overlay improvements

**Comment by jdmcbr** [comment]
Great, releasing pre-cython changes makes a lot of sense. #318 pairs well with the #267 changes, and agree on that being a nice thing to target for release. Fixing the matplotlib tests and the plots in the docs don't seem to me like things that should necessarily hold up the release, since neither of those would affect the experience of someone installing `0.3.0` (though obviously #466 has a big impact on more general usability, and is toward the top of my list of things to work on). 

I'm not sure which sjoin failures you're referencing (the only ones I'm aware of are the ones I reported recently, which were resolved by using the more recent `rtree`, so I was planning to look into whether the `requirements.test.txt` should be updated to `rtree>=0.8.3`). Regarding overlay improvements, that would be targeting #338 and #343 (or #429?)?

**Comment by jorisvandenbossche** [comment]
> I'm not sure which sjoin failures you're referencing (the only ones I'm aware of are the ones I reported recently, which were resolved by using the more recent rtree

Yes, forgot that was solved!

> Regarding overlay improvements, that would be targeting #338 and #343 (or #429?)?

Yes, all of the three (but figuring out was is happening in #343 (some simple overlay test that fail) should IMO be solved first before doing any performance optimizations)

**Comment by jorisvandenbossche** [comment]
> Fixing the matplotlib tests and the plots in the docs don't seem to me like things that should necessarily hold up the release

Fixing the matplotlib tests is somewhat release critical IMO, as currently we actually don't test that our plots work on matplotlib 2.0 (apart from people already using it on that and not really reporting issues, but still)

**Comment by jdmcbr** [comment]
> Yes, all of the three (but figuring out was is happening in #343 (some simple overlay test that fail) should IMO be solved first before doing any performance optimizations)

Agreed!

> Fixing the matplotlib tests is somewhat release critical IMO, as currently we actually don't test that our plots work on matplotlib 2.0 (apart from people already using it on that and not really reporting issues, but still)

My feeling was that allowing failures (of which there are a few) is somewhat different from not testing at all, but I see your point.

**Comment by jorisvandenbossche** [comment]
Update: I am planning to do a 0.3.0 release in the course of coming week. 
I am working on a PR that cleans-up a bit of the plotting code (will open one shortly), and then there is the defaults discussion in #318. Let's try to complete those in the coming days.

And then I propose to leave the overlay improvements for 0.4 (and it doesn't need to take another year to do a new release :-))

**Comment by jdmcbr** [comment]
@jorisvandenbossche I agree overlay improvements can wait for 0.4. Updated target then would be getting #510 and #512 merged? Anything else?

**Comment by jorisvandenbossche** [comment]
Some things to do for the release

- [ ] merge final PRs
- [ ] final update of changelog file: version (0.3.0), date, final additions
- [ ] release: empty commit and tag as 0.3.0
- [ ] push tag to github, make a release on github (edit release)
- [ ] update gh-pages
- [ ] check that the doc on readthedocs are OK
- [ ] upload source archive and wheel to pypi
- [ ] update conda-forge packages
- [ ] send announcement to mailing lists (pandas, other? seems gispython suddenly stopped being working last year) -> make list of contributors for last release

**Comment by jorisvandenbossche** [comment]
For doing the actual release, would you like to do some of the above things? (eg updating the gh-pages, doing the conda-forge update, ..)

**Comment by jdmcbr** [comment]
@jorisvandenbossche I'm happy to update gh-pages. I haven't done a conda-forge update before, but can give that a shot too. I think the changelog is in pretty good shape, though needs the addition of at least #512, and I guess noting the bug you just closed in #266. It'll be probably ~5 hours from now when I can start on anything, if that makes a difference to you.

**Comment by jorisvandenbossche** [comment]
What I can maybe do is make those last updates and actually tag, so by the time you have time, it is officially released (the tag is available), which means you can do conda-forge and the website. And then tomorrow when everything is build and available to install, I will send the announcement

**Comment by jorisvandenbossche** [comment]
For conda-forge, I will add you as a maintainer of the feedstock

**Comment by jdmcbr** [comment]
@jorisvandenbossche Sounds good!

**Comment by jorisvandenbossche** [comment]
OK, so release is pushed to github (https://github.com/geopandas/geopandas/releases/tag/v0.3.0). For conda-forge, for the url here: https://github.com/conda-forge/geopandas-feedstock/blob/master/recipe/meta.yaml#L9, it's best to update this to the url of the package I uploaded, which is https://github.com/geopandas/geopandas/releases/download/v0.3.0/geopandas-0.3.0.tar.gz (you can see how the pandas one is done: https://github.com/conda-forge/pandas-feedstock/blob/master/recipe/meta.yaml#L10)

**Comment by jorisvandenbossche** [comment]
So for conda-forge, it is basically forking the repo and making a PR where you do the needed updates in the reciples yaml file (increase the version number, update the url, update the hash, and set the build number back to 0). But now I have to go to sleep :-)

**Comment by ocefpaf** [comment]
@jdmcbr and @jorisvandenbossche it is still early on my neck of the woods :wink: 

https://github.com/conda-forge/geopandas-feedstock/pull/25

**Comment by mrocklin** [comment]
I love that Python's geospatial library is so geospatially distributed

**Comment by jdmcbr** [comment]
@jorisvandenbossche Thanks for the instructions, and thanks @ocefpaf for taking care of it anyway! The docs on gh-pages are updated; readthedocs looks okay too.

**Comment by mrocklin** [comment]
It looks like the conda-forge package was merged.  There are only two items left:

- [ ] Push to PyPI
- [ ] Broadcast by e-mail and social media

For PyPI it looks like only @kjordahl and @jorisvandenbossche have permissions.  @kjordahl would you mind increasing the pool of people who have access to PyPI?  I might recommend bumping @jorisvandenbossche up to owner status and @jdmcbr up to maintainer status.

**Comment by jorisvandenbossche** [comment]
Yes, pypi is already done, my announcement mail is ready ( but was waiting on windows conda forhe, appveyor is a bit slow)

**Comment by ocefpaf** [comment]
I would not wait for that :unamused:

**Comment by jorisvandenbossche** [comment]
They just started building I see :-) 
But announcement has been sent and tweeted. 
(sent to the pydata mailing list, if somebody knows a python gis related mailing list, you can let me know or forward it)

**Comment by jdmcbr** [comment]
Thanks all, especially @jorisvandenbossche for getting this out!

**Comment by jorisvandenbossche** [comment]
And thanks for the help!
